Jetsnack/app/src/main/java/com/example/jetsnack/ui/home/FilterScreen.kt

```diff
@@ -0,0 +1,269 @@
+            .padding(top = 12.dp, bottom = 16.dp)
+            .padding(horizontal = 4.dp)
+    ) {
+        filters.forEach { filter ->
+            FilterChip(
+                filter = filter,
+                modifier = Modifier.padding(end = 4.dp, bottom = 8.dp)
+            )
+        }
+    }
+}
+
+@Composable
+fun SortFiltersSection(sortState: String, onFilterChange: (String) -> Unit) {
+    FilterTitle(text = stringResource(id = R.string.sort))
+    Column(Modifier.padding(bottom = 24.dp)) {
+        SortFilters(
+            sortState = sortState,
+            onChanged = onFilterChange
+        )
+    }
+}
+
+@Composable
+fun SortFilters(
+    sortFilters: List<Filter> = SnackRepo.getSortFilters(),
+    sortState: String,
+    onChanged: (String) -> Unit
+) {
+
+    sortFilters.forEach { filter ->
+        SortOption(
+            text = filter.name,
+            icon = filter.icon,
+            selected = sortState == filter.name,
+            onClickOption = {
+                onChanged(filter.name)
+            }
+        )
+    }
+}
+
+@Composable
+fun MaxCalories(sliderPosition: Float, onValueChanged: (Float) -> Unit) {
+    Row {
+        FilterTitle(text = stringResource(id = R.string.max_calories))
+        Text(
+            text = stringResource(id = R.string.per_serving),
+            style = MaterialTheme.typography.body2,
+            color = JetsnackTheme.colors.brand,
+            modifier = Modifier.padding(top = 5.dp, start = 10.dp)
+        )
+    }
+    Slider(
+        value = sliderPosition,
+        onValueChange = { newValue ->
+            onValueChanged(newValue)
+        },
+        valueRange = 0f..300f,
+        steps = 5,
+        modifier = Modifier
+            .fillMaxWidth(),
+        colors = SliderDefaults.colors(
+            thumbColor = JetsnackTheme.colors.brand,
+            activeTrackColor = JetsnackTheme.colors.brand
+        )
+    )
+}
+
+@Composable
+fun FilterTitle(text: String) {
+    Text(
+        text = text,
+        style = MaterialTheme.typography.h6,
+        color = JetsnackTheme.colors.brand,
+        modifier = Modifier.padding(bottom = 8.dp)
+    )
+}
+@Composable
+fun SortOption(
+    text: String,
+    icon: ImageVector?,
+    onClickOption: () -> Unit,
+    selected: Boolean
+) {
+    Row(
+        modifier = Modifier
+            .padding(top = 14.dp)
+            .selectable(selected) { onClickOption() }
+    ) {
+        if (icon != null) {
+            Icon(imageVector = icon, contentDescription = null)
+        }
+        Text(
+            text = text,
+            style = MaterialTheme.typography.subtitle1,
+            modifier = Modifier
+                .padding(start = 10.dp)
+                .weight(1f)
+        )
+        if (selected) {
+            Icon(
+                imageVector = Icons.Filled.Done,
+                contentDescription = null,
+                tint = JetsnackTheme.colors.brand
+            )
+        }
+    }
+}
```

Comment:
Add some `@Prevew`s?

Reply:
Done

Reply:
What about adding some alternates like #549 ?

Reply:
Can you address this in another PR?